float64 final_longitude
float64 final_latitude
